tests/farmer_harvester/test_filter_prefix_bits.py

```diff
@@ -50,29 +55,58 @@ def state_changed_callback(change: str, change_data: Optional[Dict[str, Any]]) -
+    local_b_tools = await create_block_tools_async(
+        constants=test_constants.replace(NUMBER_ZERO_BITS_PLOT_FILTER=9), keychain=get_temp_keyring
     )
-    await farmer_api.new_signage_point(sp)
-    await time_out_assert(5, state_has_changed, True)
-    # We're intercepting the harvester's state changes as we're expecting
-    # a farming_info one.
-    assert state_change == "farming_info"
-    assert state_change_data is not None
-    assert state_change_data.get("eligible_plots") == eligible_plots
+    new_config = local_b_tools._config
+    local_b_tools.change_config(new_config)
+    async for harvester_services, farmer_service, _ in setup_farmer_multi_harvester(
+        local_b_tools, 1, tmp_path, local_b_tools.constants, start_services=True
+    ):
+        harvester_service = harvester_services[0]
+        assert farmer_service.rpc_server is not None
+        farmer_rpc_cl = await FarmerRpcClient.create(
+            self_hostname, farmer_service.rpc_server.listen_port, farmer_service.root_path, farmer_service.config
+        )
+        assert harvester_service.rpc_server is not None
+        harvester_rpc_cl = await HarvesterRpcClient.create(
+            self_hostname,
+            harvester_service.rpc_server.listen_port,
+            harvester_service.root_path,
+            harvester_service.config,
+        )
+        await time_out_assert(15, have_connections, True)
+        harvester_service._node.state_changed_callback = state_changed_callback
+        harvester_id = harvester_service._server.node_id
+        farmer_api = farmer_service._api
+        receiver = farmer_api.farmer.plot_sync_receivers[harvester_id]
+        if receiver.initial_sync():
+            await wait_for_plot_sync(receiver, receiver.last_sync().sync_id)
+        # This allows us to pass the plot filter with prefix bits 8 but not 9
+        challenge_hash = bytes32.from_hexstr("0x73490e166d0b88347c37d921660b216c27316aae9a3450933d3ff3b854e5831a")
+        sp_hash = bytes32.from_hexstr("0x7b3e23dbd438f9aceefa9827e2c5538898189987f49b06eceb7a43067e77b531")
+        sp = farmer_protocol.NewSignagePoint(
+            challenge_hash=challenge_hash,
+            challenge_chain_sp=sp_hash,
+            reward_chain_sp=bytes32(b"1" * 32),
+            difficulty=uint64(1),
+            sub_slot_iters=uint64(1000000),
+            signage_point_index=uint8(2),
+            peak_height=peak_height,
+        )
+        await farmer_api.new_signage_point(sp)
+        await time_out_assert(5, state_has_changed, True)
+        # We're intercepting the harvester's state changes as we're expecting
+        # a farming_info one.
+        assert state_change == "farming_info"
+        assert state_change_data is not None
+        assert state_change_data.get("eligible_plots") == eligible_plots
+        farmer_rpc_cl.close()
+        harvester_rpc_cl.close()
+        await farmer_rpc_cl.await_closed()
+        await harvester_rpc_cl.await_closed()
```

Comment:
I think this needs to live in a `finally` block, no?

Reply:
I think so, thats also why i think using fixtures makes more sense, because _i think_ pytest already takes care of it? And if not we can out the finally there without having to indent tests.

Reply:
I made a local custom fixture for this test.